features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreMapFragment.kt

```diff
@@ -0,0 +1,340 @@
+import org.dash.wallet.features.exploredash.services.UserLocationState
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+class ExploreMapFragment: SupportMapFragment() {
+    companion object {
+        private const val CURRENT_POSITION_MARKER_TAG = 0
+    }
+
+    data class MerchantMarkerUI(
+        val latitude: Double,
+        val longitude: Double,
+        val logoUrl: Bitmap
+    )
+
+    private val viewModel: ExploreViewModel by activityViewModels()
+
+    private var googleMap: GoogleMap? = null
+    private lateinit var mCurrentUserLocation: LatLng
+    private var currentAccuracy = 0.0
+    private var currentLocationMarker: Marker? = null
+    private var currentLocationCircle: Circle? = null
+
+    private var markerManager: MarkerManager? = null
+    private var markerCollection: MarkerManager.Collection? = null
+    private var circleManager: CircleManager? = null
+    private var circleCollection: CircleManager.Collection? = null
+    private var futureTarget =  mutableListOf<FutureTarget<Bitmap>>()
+    private lateinit var markersGlideRequestManager: RequestManager
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        markersGlideRequestManager = Glide.with(this)
+
+        lifecycleScope.launchWhenStarted {
+            googleMap = awaitMap()
+            showMap()
+//            setUserLocationMarkerMovementState() // TODO: why do we need to drag user location marker? That's weird
+
+            googleMap?.let { map ->
+                map.setOnCameraIdleListener {
+                    val bounds = map.projection.visibleRegion.latLngBounds
+                    viewModel.searchBounds = GeoBounds(
+                        bounds.northeast.latitude,
+                        bounds.northeast.longitude,
+                        bounds.southwest.latitude,
+                        bounds.southwest.longitude,
+                        bounds.center.latitude,
+                        bounds.center.longitude)
+                }
+            }
+        }
+
+        viewModel.currentUserLocation.observe(viewLifecycleOwner) { location ->
+            location?.let {
+                mCurrentUserLocation = LatLng(location.latitude, location.longitude)
+                currentAccuracy = location.accuracy
+                showLocationOnMap()
+            }
+        }
+
+        viewModel.searchResults.observe(viewLifecycleOwner) { results ->
+            // TODO: we probably shouldn't reset all markers but calculate the difference instead,
+            // otherwise when user pans the map, everything blinks
+            resetMap()
+
+            // TODO: For the 1st iteration of this feature, we shall limit the number of markers to be displayed
+            if (results.isNotEmpty()) {
+//                if (results.size < 20) {
+                    setMarkers(results)
+//                } else setMarkers(results.shuffled().subList(0, 20))
+            }
```

Comment:
is this displaying all markers for thousands of stores, or is results already filtered in some way to limit the number.

Reply:
The results are limited by `min(map bounds, chosen radius)`. It still means there can be a couple of thousand points shown in big cities when you zoom out. I have left a suggestion to limit this further in NMA-785.